### ISSUE TITLE
Reenable getFromDirectory / listDirectoryContents symlink following behind a flag

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#365]
Follow up from #346.

**Comment by fizzy33** [comment]
Do you want a PR for this?  Is it okay to do this as an optional parm on the getFromDirectory family of methods?

So getFrom(Browseable)Directory becomes

`
  def getFromDirectory(directoryName: String, followSymlinks: Boolean = false)(implicit resolver: ContentTypeResolver): Route =
`

Or what is the pref for how this is implemented?

**Comment by jrudolph** [comment]
Yes, a PR would be valuable. We need to keep binary compatibility so we might want to add a configuration option to `akka.http.routing` instead.

**Comment by fizzy33** [comment]
Can you point to an example of how I would read that config setting from inside of a directive?  tx

**Comment by jrudolph** [comment]
Either from the `RequestContext` if that's already available or using the `extractSettings` directive:

```scala
extractSettings.flatMap { settings =>
} // returns a new directive
```

**Comment by fizzy33** [comment]
Ha thanks for answering my question that I had deleted (the question was to point me to an example).  Great and thanks that makes me dangerous
